src/wampy.js

```diff
@@ -801,25 +798,29 @@ class Wampy {
                     this._cache.connectPromise = null;
                 }
 
-                if (onError) {
-                    onError(noSerializerAvailableError);
+                if (this._options.onError) {
+                    this._options.onError(noSerializerAvailableError);
                 }
+
+                return this._hardClose('wamp.error.protocol_violation', noSerializerAvailableError.message);
             }
+
+            this._options.serializer = new JsonSerializer();
         }
 
-        if (serializer.isBinary) {
+        if (this._options.serializer.isBinary) {
             this._ws.binaryType = 'arraybuffer';
         }
 
+        const { helloCustomDetails, authid, authmethods, authextra, realm } = this._options;
         const messageOptions = {
-            ...helloCustomDetails,
             ...this._wamp_features,
+            ...helloCustomDetails,
```

Comment:
This is bad, as there are no checks and custom options may override `wamp` feature. That must be prohibited.

Reply:
That is true 😯 I will revert it back